src/sqlfluff/core/rules/base.py

```diff
@@ -258,51 +258,55 @@ def create_after(
+    def get_fix_slices(
+        self, templated_file: TemplatedFile, within_only: bool
+    ) -> Set[RawFileSlice]:
+        """Returns slices touched by the fix."""
         # Goal: Find the raw slices touched by the fix. Two cases, based on
         # edit type:
-        # 1. "delete", "replace": Raw slices touching the anchor segment. If
-        #    ANY are templated, discard the fix.
+        # 1. "delete", "replace": Raw slices touching the anchor segment.
         # 2. "create_before", "create_after": Raw slices encompassing the two
         #    character positions surrounding the insertion point (**NOT** the
         #    whole anchor segment, because we're not *touching* the anchor
-        #    segment, we're inserting **RELATIVE** to it. If ALL are templated,
-        #    discard the fix.
+        #    segment, we're inserting **RELATIVE** to it.
         assert self.anchor.pos_marker
         anchor_slice = self.anchor.pos_marker.templated_slice
         templated_slices = [anchor_slice]
-        check_fn = any
 
+        # If "within_only" is set for a "create_*" fix, the slice should only
+        # include the area of code "within" the area of insertion, not the other
+        # side.
+        adjust_boundary = 1 if not within_only else 0
         if self.edit_type == "create_before":
             # Consider the first position of the anchor segment and the
             # position just before it.
             templated_slices = [
-                slice(anchor_slice.start, anchor_slice.start + 1),
-                slice(anchor_slice.start - 1, anchor_slice.start),
+                slice(anchor_slice.start - 1, anchor_slice.start + adjust_boundary),
             ]
-            check_fn = all
         elif self.edit_type == "create_after":
             # Consider the last position of the anchor segment and the
             # character just after it.
             templated_slices = [
-                slice(anchor_slice.stop - 1, anchor_slice.stop),
-                slice(anchor_slice.stop, anchor_slice.stop + 1),
+                slice(anchor_slice.stop - adjust_boundary, anchor_slice.stop + 1),
             ]
-            check_fn = all
         # TRICKY: For creations at the end of the file, there won't be an
         # existing slice. In this case, the function adds file_end_slice to the
         # result, as a sort of placeholder or sentinel value. We pass a literal
         # slice for "file_end_slice" so that later in this function, the LintFix
         # is interpreted as literal code. Otherwise, it could be interpreted as
         # a fix to *templated* code and incorrectly discarded.
-        fix_slices = self._raw_slices_from_templated_slices(
+        return self._raw_slices_from_templated_slices(
             templated_file,
             templated_slices,
             file_end_slice=RawFileSlice("", "literal", -1),
         )
 
-        # We have the fix slices. Now check for conflicts.
+    def has_template_conflicts(
```

Comment:
No changes in behavior for `has_template_conflicts()`, just split out `get_fix_slices()` so it could be used elsewhere.